winch/codegen/src/isa/x64/asm.rs

```diff
@@ -756,6 +756,51 @@ impl Assembler {
         });
     }
 
+    /// Compares values in src and dst and places result in dst.
+    /// Greater than operations are not supported because they are unordered.
+    pub fn xmm_cmp_rr(&mut self, src: Reg, dst: Reg, kind: FloatCmpKind, size: OperandSize) {
+        let op = match size {
+            OperandSize::S32 => SseOpcode::Cmpss,
+            OperandSize::S64 => SseOpcode::Cmpsd,
+            OperandSize::S128 => unreachable!(),
+        };
+
+        let imm = match kind {
+            FloatCmpKind::Eq => FcmpImm::Equal,
+            FloatCmpKind::Ne => FcmpImm::NotEqual,
+            FloatCmpKind::Lt => FcmpImm::LessThan,
+            FloatCmpKind::Le => FcmpImm::LessThanOrEqual,
+            // There are UnorderedGreaterThan operations available but they're
```

Comment:
I'm not sure this is the behaviour we want. The behaviour of the unordered comparison initially surprised me. But maybe there's a legitimate reason we would want to support unordered comparison operations in the microassembler now as opposed to adding that behaviour later.